junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/MethodSelectorResolver.java

```diff
@@ -36,4 +36,28 @@ private void resolve(MethodSelector selector, TestClassCollector collector) {
 		collector.addFiltered(testClass, adapter(matchMethodDescription(methodDescription)));
 	}
 
+	private static Filter matchMethodDescription(final Description desiredDescription) {
+		return new Filter() {
+			@Override
+			public boolean shouldRun(Description description) {
+				if (description.isTest()) {
+					return desiredDescription.equals(description)
+							|| description.getMethodName().startsWith(desiredDescription.getMethodName() + "[");
```

Comment:
Please add an inline comment explaining the additional logic that accounts for parameterized tests.